File: vector/src/vector/Mesh.java
package vector;

public class Mesh {
	public Vector3[] vertices;
	public int[][] patches;
	
	public Mesh() {
		this.vertices = null;
		this.patches = null;
	}
	
	public Mesh(Vector3[] vertices, int[][] patches) {
		this.vertices = vertices;
		this.patches = patches;
	}
}

File: VectorWorld/mesh.cpp
#include "main.h"

Mesh::Mesh() {
    vertices = NULL;
    patches = NULL;
}

Mesh::Mesh(int numVertices, Vector3 *vertices, int numPatches, int **patches) {
    this->numVertices = numVertices;
    this->vertices = vertices;
    
    this->numPatches = numPatches;
    this->patches = patches;
}

Mesh::~Mesh() {
    delete [] vertices;
    vertices = NULL;
    
    for (int i = 0; i < numPatches; i++) {
        delete [] patches[i];
    }
    delete [] patches;
    patches = NULL;
}

int Mesh::getNumVertices() {
    return numVertices;
}

int Mesh::getNumPatches() {
    return numPatches;
}